### DISCUSSION TITLE
Preferred pattern for high-volume backfill runs

### DISCUSSION THREAD

**Discussion by JoeNaso** [dagster-io/dagster#18476] [Q&A]
What is the preferred pattern for launching "high" volume backfills? I'll use 365 days worth of hourly partitions as an example. 

If we're processing data from multiple sources, and we want to generate hourly partitions per day, you wind up with a relatively large number of job runs: 

- 2 sources: 17,500+
- 3 sources: 26,200+
- ... etc 

I (naively? ambitiously?) tried to run a backfill like this and cause some degradation of the webserver and UI. In the case that we **must** support hourly partitions, what is the best pattern to backfill at this scale? 

Should it be run via CLI? Adding records via GraphQL? Something else? 

There dont appear to be limitations on the scope of backfills you can trigger from the UI, which is a related but separate point. If it's preferrable to batch these larger backfill runs, it would be nice to have a that limitation known.

**Comment by sryza** [comment]
Have you looked at `BackfillPolicy`s?
- https://docs.dagster.io/concepts/partitions-schedules-sensors/backfills#launching-single-run-backfills-using-backfill-policies-
- https://docs.dagster.io/_apidocs/partitions#backfill-policy-experimental

**Reply by JoeNaso** [reply]
This doesnt completely address my concern. The policy still launches as many jobs, based on my testing.  If I launch a high volume number of jobs, there is UI degradation and it's difficult to observe the job run status because of that. Partition loading fails in the UI, as well.

Is that volume of queued runs simply too high? What is the suggested approach for launching a large number of backfills?

**Reply by j-blackwell** [reply]
Hi @sryza and @JoeNaso,

I have 2 related issues. I am using a `BackfillPolicy.single_run()` on each asset within a job. 

## Issue 1
The data processing takes 8 seconds to get to `STEP_OUTPUT`, but each `ASSET_MATERIALISATION` is logged individually, causing the asset to take 28 mins for 12k partitions to finish. Is it possible to combine all these materialisation results into a single batch?

![image](https://github.com/dagster-io/dagster/assets/33688964/31f1a152-53dd-4531-8cbf-4f8a30bf5a71)

## Issue 2
When I run the job as a whole (or even multiple asset materialisations at the same click), dagster ignores the `BackfillPolicy`, and tries to trigger several thousand job runs. As a result, there is no way to execute the whole job in a single backfill. Is there any plan to add `BackfillPolicy` to jobs as well?

But this means that each asset has to be backfilled individually and because the materialisation results are not posted together, dagster doesn't think that any upstream assets have been fully materialised until all partitions have logged their materialisations. This means that it is a 30 minute wait for each layer of dependencies before I can trigger the next one, and the result is that a computation that takes ~30seconds to complete will take ~2hours in dagster (4 layers of dependencies). 

##
Am I doing something wrong here? Because overall, I would say that this functionality is pretty integral as it will surely appear in any dagster migration project or any project that uses partitioned assets with historical data, and I am not sure why I can't see more discussion on it.

So I have created a separate issue (#18789)
